client/components/site-selector/index.jsx

```diff
@@ -431,15 +447,22 @@ const SiteSelector = React.createClass( {
 			</div>
 		);
 	}
-} );
+}
 
 export default connect( ( state ) => {
 	const user = getCurrentUser( state );
 	const visibleSiteCount = get( user, 'visible_site_count', 0 );
+
 	return {
+		sites: getSites( state ),
 		showRecentSites: get( user, 'visible_site_count', 0 ) > 11,
 		recentSites: getPreference( state, 'recentSites' ),
 		siteCount: get( user, 'site_count', 0 ),
 		visibleSiteCount: visibleSiteCount,
+		getSite: getSite.bind( null, state ),
```

Comment:
@blowery: correct, and it's been a priority of mine to tackle these. There were other such occurrences that we got rid of in the meantime, but this one stayed. Now that this is merged, it unblocks some other work around `SiteSelector`, `Site`, etc., which seeks to make more parts only care about site IDs and not need to satisfy as much data. /cc @ockham